tests/blockchain/test_blockchain.py

```diff
@@ -3634,3 +3634,46 @@ async def test_reorg_flip_flop(empty_blockchain, bt):
 
     for block in chain_b[40:]:
         await _validate_and_add_block(b, block)
+
+
+@pytest.mark.parametrize("unique_plots_window", [1, 2])
+@pytest.mark.parametrize("bt_respects_soft_fork3", [True, False])
+@pytest.mark.parametrize("soft_fork3_height", [0, 10, 10000])
+@pytest.mark.asyncio
+async def test_soft_fork3_activation(
+    blockchain_constants, bt_respects_soft_fork3, soft_fork3_height, db_version, unique_plots_window
+):
+    with TempKeyring() as keychain:
+        bt = await create_block_tools_async(
+            constants=blockchain_constants.replace(
+                SOFT_FORK3_HEIGHT=(0 if bt_respects_soft_fork3 else 10000),
+                UNIQUE_PLOTS_WINDOW=unique_plots_window,
+            ),
+            keychain=keychain,
+        )
+        blockchain_constants = bt.constants.replace(SOFT_FORK3_HEIGHT=soft_fork3_height)
+        b, db_wrapper, db_path = await create_blockchain(blockchain_constants, db_version)
+        blocks = bt.get_consecutive_blocks(25)
+        for height, block in enumerate(blocks):
+            await _validate_and_add_block_multi_error_or_pass(b, block, [Err.INVALID_POSPACE])
+            peak = b.get_peak()
+            assert peak is not None
+            if peak.height != height:
+                break
+
+        peak = b.get_peak()
+        assert peak is not None
+
+        if bt_respects_soft_fork3 or unique_plots_window == 1:
+            assert peak.height == 24
```

Comment:
I think it would be good to add some comments here, to explain these values we expect. why is this 24?